arch/inst/C/c.sd.yaml

```diff
@@ -35,6 +47,15 @@ operation(): |
     raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
   }
 
-  XReg virtual_address = X[creg2reg(rs1)] + imm;
+  XReg virtual_address = X[creg2reg(xs1)] + imm;
```

Comment:
So, the `creg2reg` is only necessary in 32bit mode, where the register field "xs1" is only 3 bits and has to be transformed into the proper register index. This is much like what you've already done for `c.ld`.